### ISSUE TITLE
use of genericModelSubstitutes

### ISSUE THREAD

**Issue by ctamisier** [springfox/springfox#347]
Hi,
If I do this:

```
@RequestMapping(value = "/test3", method = RequestMethod.GET)
@ResponseBody
public ResponseEntity<Resource<Album>> test3() {
         return null;
}
```

configured with 

```
@Bean
public SwaggerSpringMvcPlugin swaggerSpringMvcPlugin(SpringSwaggerConfig springSwaggerConfig) {
        return new SwaggerSpringMvcPlugin(springSwaggerConfig)
.genericModelSubstitutes(ResponseEntity.class, Resource.class);
}
```

the json is:

```
"path": "/music/test3",
        "description": "test3",
        "operations": [{
            "method": "GET",
            "summary": "test3",
            "notes": "test3",
            "type": "Resource«Album»",
            "nickname": "test3",
            "parameters": [],
```

**"type": "Resource«Album»"**, I was expecting to have **Album** because Resource is also a genericModelSubstitute

So in the UI (http://localhost:9080/spring3-music/sdoc.jsp#!/albums/test3) there is no "Model" associated.

You can test here: https://github.com/ctamisier/swagger-springmvc-demo/commit/0bf4ead2b9fff6c538d6ea806646ed74cfa01232
could it be a bug ?
Thanks for your help.

**Comment by rlogiacco** [comment]
I'm using version 2.4.0 and a `List<Resource<Article>>` with `Resource.class` being added as `genericModelSubstitutes` outputs:
```
200: {description: "OK", schema: {type: "array", items: {$ref: "#/definitions/Resource«Article»"}}
```

But another endpoint returning `Resource<Article>` outputs:
```
200: {description: "OK", schema: {$ref: "#/definitions/Article"}}
```

Am I missing something?

**Comment by dilipkrish** [comment]
Would you mind trying with `2.6.1`. Pretty sure this bug is fixed @rlogiacco

**Comment by rlogiacco** [comment]
@dilipkrish I updated to `2.6.1` and also added a `forCodeGeneration(true)` just to ensure the update went through correctly (you never know) and I now get:
```
200: {description: "OK", schema: {type: "array", items: {$ref: "#/definitions/ResourceOfAgreement"}}}
```

Apparently the bug is still there in my case.
What I noticed trying to debug the thing is that `List` seems not being returning `true` when `hasWildcards(...)` is executed... I'm looking at a breakpoint placed at line 49 of `springfox.documentation.schema.AlternateTypeRule.alternateFor(ResolvedType)`...

In other words it seems the problem, in this case, lies on the outer `List` element which doesn't allow the resolution to start... Any advice?

**Comment by dilipkrish** [comment]
Oh yeah I forgot. You need to add an alternate type rule for that explicitly. A rule that replaces List<Resource<WildcardType> with List<WildcardType>.

```java

newRule(
    resolver.resolve(List.class, resolver.resolve(Resource.class, WildcardType.class)),
    resolver.resolve(List.class, WildcardType.class));

```
